pkg/asset/installconfig/nutanix/nutanix.go

```diff
@@ -0,0 +1,304 @@
+	nutanixtypes "github.com/openshift/installer/pkg/types/nutanix"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+type PrismCentralClient struct {
+	PrismCentral string
+	Username     string
+	Password     string
+	Port         string
+	V3Client     *nutanixClientV3.Client
+	Insecure     bool
+}
+
+// Platform collects Nutanix-specific configuration.
+func Platform() (*nutanix.Platform, error) {
+	nutanixClient, err := getClients()
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.TODO()
+	v3Client := nutanixClient.V3Client
+	peUUID, err := getPrismElement(ctx, v3Client)
+	if err != nil {
+		return nil, err
+	}
+	subnetUUID, err := getSubnet(ctx, v3Client, peUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	apiVIP, ingressVIP, err := getVIPs()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get VIPs")
+	}
+
+	platform := &nutanix.Platform{
+		PrismCentral:     nutanixClient.PrismCentral,
+		Insecure:         nutanixClient.Insecure,
+		Port:             nutanixClient.Port,
+		Username:         nutanixClient.Username,
+		Password:         nutanixClient.Password,
+		PrismElementUUID: peUUID,
+		SubnetUUID:       subnetUUID,
+		APIVIP:           apiVIP,
+		IngressVIP:       ingressVIP,
+	}
+	return platform, nil
+
+}
+
+// getClients() surveys the user for username, password, port & prism central.
+// Validation on the three fields is performed by creating a client.
+// If creating the client fails, an error is returned.
+func getClients() (*PrismCentralClient, error) {
+	var prismCentral, username, password, port string
+	var secure bool
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Prism Central",
+				Help:    "The domain name or IP address of the Prism Central to be used for installation.",
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				return validate.Host(ans.(string))
+			}),
+		},
+	}, &prismCentral); err != nil {
+		return nil, errors.Wrap(err, "failed UserInput")
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Username",
+				Help:    "The username to login to the Prism Central.",
+			},
+			Validate: survey.Required,
+		},
+	}, &username); err != nil {
+		return nil, errors.Wrap(err, "failed UserInput")
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "Password",
+				Help:    "The password to login to Prism Central.",
+			},
+			Validate: survey.Required,
+		},
+	}, &password); err != nil {
+		return nil, errors.Wrap(err, "failed UserInput")
+	}
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Port",
+				Help:    "The port used to to login to the Prism Central.",
+				Default: "9440",
+			},
+			Validate: survey.Required,
+		},
+	}, &port); err != nil {
+		return nil, errors.Wrap(err, "failed UserInput")
+	}
+	if err := survey.Ask([]*survey.Question{
```

Comment:
may need to remove this based on the outcome of https://github.com/openshift/installer/pull/5603#discussion_r810971769